crates/bevy_ecs/src/event/read_iterators.rs

```diff
@@ -233,6 +235,10 @@ impl<'a, E: Event> EventParIter<'a, E> {
                     });
                 }
             });
+
+            // If we get to hear we're guaranteed to have seen all events
```

Comment:
Left a suggestion for this comment [here](https://github.com/bevyengine/bevy/pull/13836#discussion_r1639621894).